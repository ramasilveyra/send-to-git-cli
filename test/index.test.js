describe('while using send-static-cli', () => {
  let sendToGit = null;
  let sendToGitCLI = null;

  beforeAll(async () => {
    sendToGit = (await import('send-to-git')).default;
    jest.mock('send-to-git');
    sendToGitCLI = (await import('../src/index')).default;
  });

  beforeEach(() => {
    sendToGit.mockReset();
    sendToGit.mockImplementation(() => Promise.resolve());
  });

  afterAll(() => {
    jest.unmock('send-to-git');
  });

  it('should correctly handle required options', () => {
    sendToGitCLI([ '-s', './some-folder', '-d', './public_html', '-r', 'git@github.com:ramasilveyra/send-to-git-cli.git' ]);
    expect(sendToGit.mock.calls.length).toEqual(1);
    expect(sendToGit.mock.calls[0]).toMatchSnapshot();
  });

  it('should log version with "-v" or "--version"', () => {
    /* eslint-disable no-console */
    process.exit = jest.fn();
    console.log = jest.fn();
    sendToGitCLI(['-v']);
    sendToGitCLI(['--version']);
    expect(process.exit.mock.calls.length).toEqual(2);
    expect(console.log.mock.calls.length).toEqual(2);
    expect(console.log.mock.calls[0]).toMatchSnapshot();
    expect(console.log.mock.calls[1]).toMatchSnapshot();
  });
});
