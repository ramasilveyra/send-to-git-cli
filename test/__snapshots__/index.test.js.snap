// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`while using send-static-cli should correctly handle required options 1`] = `
Array [
  "./some-folder",
  "./public_html",
  "git@github.com:ramasilveyra/send-to-git-cli.git",
  Object {
    "branch": "master",
    "commitMessage": "Release",
  },
]
`;

exports[`while using send-static-cli should log version with "-v" or "--version" 1`] = `
Array [
  "1.0.0",
]
`;

exports[`while using send-static-cli should log version with "-v" or "--version" 2`] = `
Array [
  "1.0.0",
]
`;
